src/data_loader.py

```diff
@@ -0,0 +1,160 @@
+        self.replacement = replacement
+        # images_list is the name 
+        self.images_list = list(image_dict["ships"].keys()) + image_dict["w/ships"]
+        random.shuffle(self.images_list) 
+        self.transform = transform
+    
+    def __len__(self):
+        return len(self.images_list)
+    
+    def rle_decode(self, mask_rle, shape=(768, 768)):
+        """
+        mask_rle: run-length as string formated (start length)
+        shape: (height,width) of array to return 
+        Returns numpy array, 1 - mask, 0 - background
+        """
+        s = mask_rle.split()
+        starts, lengths = [np.asarray(x, dtype=int) for x in (s[0:][::2], s[1:][::2])]
+        starts -= 1
+        ends = starts + lengths
+        img = np.zeros(shape[0]*shape[1], dtype=np.uint8)
+        for lo, hi in zip(starts, ends):
+            img[lo:hi] = 1
+        return img.reshape(shape).T  # Needed to align to RLE direction
+    
+    def masks_as_image(self, in_mask_list):
+        # Take the individual ship masks and create a single mask array for all ships
+        all_masks = np.zeros((768, 768), dtype = np.int16)
+        #if isinstance(in_mask_list, list):
+        for mask in in_mask_list:
+            if isinstance(mask, str):
+                all_masks += self.rle_decode(mask)
+        return np.expand_dims(all_masks, -1)
+    
+    
+    def retrive_image_matrix(self, image_id):
+        # google platform (TODO!!!!!!!!!)
+        image_path = path.join(path.dirname(__file__), '../data/train_v2/', image_id)
+
+        # kaggle image path
+        #image_path = "/kaggle/input/airbus-ship-detection/train_v2/" + image_id
+        img = PIL.Image.open(image_path)
+        image_mat = np.array(img)
+        return image_mat
+        
+    def __getitem__(self, idx):
+        image_id = self.images_list[idx]
+        no_ship_list = image_dict["w/ships"]
+        image_mat = self.retrive_image_matrix(image_id)
+        # dataloader for classification
+        if self.loader_type == "classification":
+            if image_id in no_ship_list:
+                sample = {"image": image_mat, "class": 0}
+            else:
+                sample = {"image": image_mat, "class": 1}
+        
+        # dataloader for segmentation 
+        else: 
+            mask_list = image_dict["ships"][image_id]
+            mask = self.masks_as_image(mask_list) 
+            sample = {"image": image_mat, "mask": mask}
+
+        
+        # remove image id if sample without replacement
+        if self.replacement:
+            self.images_list.remove(image_id)
+
+        if self.transform:
+            sample = self.transform(sample)
+        
+        return sample
+    
+    def __random_sampler__(self):
+        n_image_list = len(self.images_list)
+        
+        # randomly choose a idx
+        random_idx = random.randint(0,n_image_list)
+        sample = self.__getitem__(random_idx)
+        return sample
+
+
+
+# helper functions
```

Comment:
Let's create a file called something like "helper.py" (if you can come up with better name, I'm down :D ) and move these functions there. They shouldn't be in dataloader file really